### ISSUE TITLE
BUG - Unable to install server due to GraphQL version conflicts

### ISSUE THREAD

**Issue by celinepelletier** [overture-stack/arranger#736]
Dependancy conflicts for GraphQL in arranger/server

## Description
I try to run a npm install @arranger/server but it fails because it's not able to resolve dependancies due to conflicts 

## Expected Behaviour
It should be able to install @arranger/server without any dependancies conflicts

## Actual Behaviour
```
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@14.7.0
npm WARN node_modules/graphql
npm WARN   graphql@"^14.5.3" from @arranger/server@2.15.0
npm WARN   node_modules/@arranger/server
npm WARN     @arranger/server@"2.15.0" from the root project
npm WARN   9 more (@arranger/admin, @arranger/mapping-utils, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.13.2" from graphql-middleware@1.3.1
npm WARN node_modules/graphql-middleware
npm WARN   graphql-middleware@"1.3.1" from @arranger/schema@2.15.0
npm WARN   node_modules/@arranger/schema
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@14.7.0
npm WARN node_modules/graphql
npm WARN   graphql@"^14.5.3" from @arranger/server@2.15.0
npm WARN   node_modules/@arranger/server
npm WARN     @arranger/server@"2.15.0" from the root project
npm WARN   9 more (@arranger/admin, @arranger/mapping-utils, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.7.2" from graphql-scalars@0.1.5
npm WARN node_modules/graphql-scalars
npm WARN   graphql-scalars@"^0.1.5" from @arranger/schema@2.15.0
npm WARN   node_modules/@arranger/schema
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: graphql@0.13.2
npm ERR! node_modules/graphql
npm ERR!   peer graphql@"^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-server-core@2.25.2
npm ERR!   node_modules/apollo-server-core
npm ERR!     apollo-server-core@"^2.14.2" from @arranger/server@2.15.0
npm ERR!     node_modules/@arranger/server
npm ERR!       @arranger/server@"2.15.0" from the root project
npm ERR!     apollo-server-core@"^2.25.2" from apollo-server-express@2.25.2
npm ERR!     node_modules/apollo-server-express
npm ERR!       apollo-server-express@"^2.14.2" from @arranger/server@2.15.0
npm ERR!       node_modules/@arranger/server
npm ERR!         @arranger/server@"2.15.0" from the root project
npm ERR!       2 more (@arranger/admin, apollo-server)
npm ERR!     1 more (apollo-server)
npm ERR!   peer graphql@"^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-server-express@2.25.2
npm ERR!   node_modules/apollo-server-express
npm ERR!     apollo-server-express@"^2.14.2" from @arranger/server@2.15.0
npm ERR!     node_modules/@arranger/server
npm ERR!       @arranger/server@"2.15.0" from the root project
npm ERR!     apollo-server-express@"^2.14.2" from @arranger/admin@2.15.0
npm ERR!     node_modules/@arranger/admin
npm ERR!       @arranger/admin@"^2.15.0" from @arranger/server@2.15.0
npm ERR!       node_modules/@arranger/server
npm ERR!         @arranger/server@"2.15.0" from the root project
npm ERR!     1 more (apollo-server)
npm ERR!   19 more (apollo-link-http, apollo-server, graphql-tools, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer graphql@"^14.5.3" from @arranger/mapping-utils@2.15.0
npm ERR! node_modules/@arranger/mapping-utils
npm ERR!   @arranger/mapping-utils@"^2.15.0" from @arranger/server@2.15.0
npm ERR!   node_modules/@arranger/server
npm ERR!     @arranger/server@"2.15.0" from the root project
npm ERR!   @arranger/mapping-utils@"^2.15.0" from @arranger/admin@2.15.0
npm ERR!   node_modules/@arranger/admin
npm ERR!     @arranger/admin@"^2.15.0" from @arranger/server@2.15.0
npm ERR!     node_modules/@arranger/server
npm ERR!       @arranger/server@"2.15.0" from the root project
npm ERR!   1 more (@arranger/schema)
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

## Possible Fix
Upgrade graphql-middleware version in @arranger/schema since currently it's fixed to 1.3.1 which is a very old version and it needs an old graphql version 0.13.X whereas others want graphql 14.X

## Steps to Reproduce
Run `npm install @arranger/server`

## Your Environment
Node 16 with npm 7

I also tried with node 14 npm 6, install is able to finish but wit this warnings : 

```
npm WARN graphql-scalars@0.1.5 requires a peer of graphql@^0.7.2 but none is installed. You must install peer dependencies yourself.
npm WARN graphql-middleware@1.3.1 requires a peer of graphql@^0.13.2 but none is installed. You must install peer dependencies yourself.
npm WARN graphql-tools@3.1.1 requires a peer of graphql@^0.13.0 but none is installed. You must install peer dependencies yourself.
```

Which show the same issue actually

**Comment by caravinci** [comment]
Bonjour, @celinepelletier! Hope you've had a great weekend.

A few minutes ago I've opened a PR to try and resolve this issue (https://github.com/overture-stack/arranger/pull/737), which long story short, should resolve the problem you describe using NodeJS v13 (the current, fully supported version).
Please feel free to review it, so we can try publishing, and then applying those changes in your instance.

While I'm confident that should do the trick, as mentioned over slack, last week I've started a test branch that updates support to v16. If the PR mentioned above doesn't solve the issue, we can try using that branch, with the caveat that it hasn't been fully tested and is not ready for production.

**Comment by caravinci** [comment]
Reopening this ticket, as it got closed earlier from merging the connected PR.

I've published the changes to npm under tag `2.16.0`. 
Would you mind please trying it out, @celinepelletier?
If this doesn't work, we can try a different workaround to get your implementation unblocked while the rewrite is complete.

Thank you!

**Comment by caravinci** [comment]
Hm that's unfortunate. I was hoping updating the locks the way i did would prevent the error from happening, even if getting those warnings (which will disappear too, after the rewrite is finished) 
Guess for the time being, if it works for you and just so you're not blocked waiting for a fix, you can probably run:
`npm i @arranger/server@2.16.0 --legacy-peer-deps` while on N16, which should avoid the error.
